### DISCUSSION TITLE
What is the best practice for writing parquet for export?

### DISCUSSION THREAD

**Discussion by seandavi** [ClickHouse/ClickHouse#49416] [Q&A] [answered]
I have a 40GB table in clickhouse and want to export to parquet. The table isn't partitioned. What are the best approaches for writing out to Parquet? It seems clickhouse supports writing the table as a single large parquet file. Is there a way to automatically split that file into, for example, 1GB (or 1M rows) parquet files? I can certainly pull data out into python or whatnot and write from there, but I wondered if splitting such large tables on export is supported natively.

Thanks.

**Comment by den-crane** [comment]
No such feature. 
you may try https://github.com/ClickHouse/ClickHouse/issues/39081#issuecomment-1342814892

**Comment by seandavi** [comment, marked as answer]
I ended up using this technique: https://clickhouse.com/docs/en/integrations/s3#splitting-large-files

It works just fine for an export. Each export overwrites the previous. If relying on globbing downstream, consider cleaning the export objects from previous exports first.

**Reply by alanpaulkwan** [reply]
Is there such a way without s3? I can always use minio / s3 bucket but what if writing to local?

**Reply by alexey-milovidov** [reply]
@alanpaulkwan INSERT INTO FUNCTION file(...) PARTITION BY ...

**Reply by alanpaulkwan** [reply]
Amazing!
